packages/headless/src/app/search-engine/search-engine.test.ts

```diff
@@ -26,20 +25,58 @@ describe('searchEngine', () => {
       initEngine();
     });
 
-    it('passing an invalid searchHub throws', () => {
-      options.configuration.search!.searchHub = '';
-      expect(initEngine).toThrow();
-    });
+    describe('validating the search configuration', () => {
+      it('passing an empty pipeline does not throw', () => {
+        options.configuration.search!.pipeline = '';
+        expect(initEngine).not.toThrow();
+      });
+
+      it('passing an empty searchHub throws', () => {
```

Comment:
What ends up throwing here is the `updateSearchConfiguration` action creator (https://github.com/coveo/ui-kit/pull/4339/files#diff-21cfc0c7d2b75f9d016dfb8c09b9322767bd002b34634cca40fdd8058252058fR111).

Other than replacing the `apiBaseUrl` parameter with `proxyBaseUrl` and modifying the validation for that param, this PR doesn't change the past validation behavior. I did, however, add some missing tests!

I agree that the test descriptions could be clearer, something like `passing an empty string for the searchHub throws` (and similar for `locale` and `timezone`).

Since this is by no means urgent and I don't want to have to go through the whole CI again before merging this PR, I'll do this in another Jira.

Reply:
See [KIT-3568](https://coveord.atlassian.net/browse/KIT-3568)